### ISSUE TITLE
Support for HttpClientConfigBuilderCustomizer

### ISSUE THREAD

**Issue by robmoore** [VanRoy/spring-data-jest#33]
In order to avoid 401 redirects on every request, it's possible to do pre-preemptive authentication with Jest using HttpClientConfigBuilderCustomizer.

https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-nosql.html#boot-features-connecting-to-elasticsearch-jest

I don't see any support for this in spring-data-jest and was wondering if it's something you would consider supporting.

**Comment by robmoore** [comment]
One workaround is to exclude `ElasticsearchJestAutoConfiguration` so that package `org.springframework.boot.autoconfigure.elasticsearch.jest.JestAutoConfiguration` is picked up instead. 
```
@SpringBootApplication(exclude = {ElasticsearchAutoConfiguration.class, ElasticsearchDataAutoConfiguration.class, ElasticsearchJestAutoConfiguration.class})
```
The downside of this is that you don't get an embedded version of Elasticsearch for free which is useful for testing.

**Comment by VanRoy** [comment]
@robmoore Indeed, it's not supported at this time. It could be a nice addons. I will investigate on it.

**Comment by aminmc** [comment]
@VanRoy Are you investigating this? If not I'm happy to help out. 

Cheers

**Comment by VanRoy** [comment]
Hi @aminmc , Yes i think we can integrate this feature in the next 3.x version. Please feel free to push a PR with this enhancement, it's a pleasure for me to get your help.

**Comment by aminmc** [comment]
@VanRoy cool. I'll get on the case 👍

**Comment by VanRoy** [comment]
Resolved by #36

**Comment by VanRoy** [comment]
Release 2.3.2 with your HttpClientConfigBuilderCustomizer support.

Thanks for your help.
